src/core/data.jl

```diff
@@ -7,10 +7,31 @@ function update_data!(data::Dict{String,<:Any}, new_data::Dict{String,<:Any})
     else
         Memento.warn(_LOGGER, "running update_data with data that does not include per_unit field, units may be incorrect")
     end
+
     _update_data!(data, new_data)
 end
 
 
+function modify_data_with_function!(data::Dict{String,<:Any}, it::String, modification_function::Function; apply_to_nws::Bool = true)
+    data_it = ismultiinfrastructure(data) ? data["it"][it] : data
+
+    if ismultinetwork(data_it) && apply_to_nws
+        for (nw, nw_data) in data_it["nw"]
+            modification_function(nw_data)
+        end
+    else
+        modification_function(data_it)
+    end
+end
+
+
+function get_data_with_function(data::Dict{String, <:Any}, it::String, getter_function::Function)
```

Comment:
For this function I have a feeling similar to `modify_data_with_function!`.   I am thinking we call it something like `get` or `get_data` (kind of like the current `update_data`), or maybe even `apply` (which would be distinct from the `apply_to_all!` by only applying to exactly one time and throw an error otherwise).

For consistency with `apply_to_all!` I would also add an optional kwarg for `multinetwork_function::Bool = false` and then do the network lookup in a similar way to the infrastructure type.

In terms of function arguments, I would suggest `data, getter_function, it` (as per previous discussion) and consider the name `getter` in place of `getter_function` as the argument type provides the "function" information.

Reply:
I agree with everything said here. Let's use `get_data` since this function should always return something, and because `get` is a Julia-native function.